commands/utils/packagehandlers/packagehandlers_test.go

```diff
@@ -212,27 +229,41 @@ func TestYarnPackageHandler_UpdateDependency(t *testing.T) {
 		{
 			vulnDetails: &utils.VulnerabilityDetails{
 				SuggestedFixedVersion:       "1.2.6",
+				IsDirectDependency:          false,
+				VulnerabilityOrViolationRow: formats.VulnerabilityOrViolationRow{Technology: coreutils.Yarn, ImpactedDependencyName: "minimist"},
+			},
+			fixSupported: false,
+			projectDir:   "yarn",
+		},
+		{
+			vulnDetails: &utils.VulnerabilityDetails{
+				SuggestedFixedVersion:       "1.2.6",
+				IsDirectDependency:          true,
 				VulnerabilityOrViolationRow: formats.VulnerabilityOrViolationRow{Technology: coreutils.Yarn, ImpactedDependencyName: "minimist"},
-			}, fixSupported: false,
+			},
+			fixSupported: true,
+			projectDir:   "yarn1",
 		},
 		{
 			vulnDetails: &utils.VulnerabilityDetails{
 				SuggestedFixedVersion:       "1.2.6",
 				IsDirectDependency:          true,
 				VulnerabilityOrViolationRow: formats.VulnerabilityOrViolationRow{Technology: coreutils.Yarn, ImpactedDependencyName: "minimist"},
-			}, fixSupported: true,
+			},
+			fixSupported: true,
+			projectDir:   "yarn2",
 		},
 	}
 	for _, test := range testcases {
 		t.Run(test.vulnDetails.ImpactedDependencyName+" direct:"+strconv.FormatBool(test.vulnDetails.IsDirectDependency), func(t *testing.T) {
```

Comment:
t.Run ( X , func ....

X is the name of the sub test, in your case it should contain the project dir to separate between the yarn1 and 2 tests